scanner/mappers/mappers.go

```diff
@@ -517,17 +518,12 @@ func convertMapToSlice[IN any, OUT any](convF func(*IN) *OUT, in map[string]*IN)
 
 // fixedInVersion returns the fixed in string, typically provided the report's
 // `FixedInVersion` as a plain string, but, in some OSV updaters, it can be an
-// urlencoded string or expected to be extracted from Range.Upper.
+// urlencoded string.
 func fixedInVersion(v *claircore.Vulnerability) string {
 	fixedIn := v.FixedInVersion
-	if fixedIn == "" && v.Range != nil {
-		// If fixed in is empty but range is provided, use that.
-		fixedIn = v.Range.Upper.String()
-	} else {
-		// Try to parse url encoded params; if expected values are not found leave it.
-		if q, err := url.ParseQuery(fixedIn); err == nil && q.Has("fixed") {
-			fixedIn = q.Get("fixed")
-		}
```

Comment:
Missing clarification why not use Range.

Reply:
The upper part of the Range only tells you the last known affected version. It doesn't necessarily tell you which version fixes the vulnerability.

Also, Quay only looks at the `FixedInVersion`: https://github.com/quay/quay/blob/master/data/secscan_model/secscan_v4_model.py

Lastly, I don't think the `Range` is sent over in the `VulnerabilityReport` anyway lol: https://github.com/quay/claircore/blob/v1.5.23/datastore/postgres/get.go#L90

Reply:
Quay's logic is essentially the same as this: https://github.com/quay/quay/blob/master/data/secscan_model/secscan_v4_model.py#L111

I find it comforting to know that a lot of our assumptions/implementation details were also done independently by Quay. Another example being that Quay also only cares about the first environment: https://github.com/quay/quay/blob/master/data/secscan_model/secscan_v4_model.py#L583

Reply:
> The upper part of the Range only tells you the last known affected version. It doesn't necessarily tell you which version fixes the vulnerability.

If the upper bound is open, it means that the version in the upper range is the one that fixes the vuln, no?

Example: https://github.com/advisories/GHSA-257q-pv89-v3xv

Also, if you look in ClairCore's [OSV updater](https://github.com/quay/claircore/blob/b62d484f357d422a450e6cb2258dac861c23f2d0/updater/osv/osv.go#L554-L558) you will notice that the upper range is set to the fixed version. If not fixed, then it's set to "+Inf" (int max).

In tests, the vuln database showed [empty fixed by, but ranges with upper bounds](https://redhat-internal.slack.com/archives/C020W1C61G8/p1701999678614669?thread_ts=1701999672.995029&cid=C020W1C61G8).

> Lastly, I don't think the Range is sent over in the VulnerabilityReport anyway lol

Ah, if that's the case, then it does make sense to remove Range. Feel free to resolve.

> Quay's logic

I think following the steps of Quay is the safest move.